src/scheduler/tasks/CheckOfflineChargingStationsTask.ts

```diff
@@ -26,14 +27,27 @@ export default class CheckOfflineChargingStationsTask extends SchedulerTask {
           offlineSince
         }, Constants.DB_PARAMS_MAX_LIMIT);
         if (chargingStations.count > 0) {
-          const chargingStationIDs: string = chargingStations.result.map((chargingStation) => chargingStation.id).join(', ');
-          // Send notification
-          await NotificationHandler.sendOfflineChargingStations(
-            tenant.id, {
+          for (let i = 0; i < chargingStations.result.length; i++) {
+            const chargingStationClient = await ChargingStationClientFactory.getChargingStationClient(tenant.id, chargingStations.result[i]);
+            if ((await chargingStationClient.getConfiguration({})).configurationKey.length > 0) {
+              // Update Heartbeat
+              chargingStations.result[i].lastHeartBeat = new Date();
+              // Save
+              await ChargingStationStorage.saveChargingStation(tenant.id, chargingStations.result[i]);
+              // Remove from notification
+              chargingStations.result.splice(i);
```

Comment:
Did you test that code ?
Because splice will remove the record from the list and the next iteration will miss the next charging station!
I would expect you start from the end:
```
      for (let i = chargingStations.result.length - 1; i >= 0; i--) {
        // Remove
        chargingStations.result.splice(i, 1);
      }

```
And that you test each line of code in the debugger.
Thanks.

Reply:
You are right.
I tested it in the debugger for the slf and cah tenant and after some changes it does what it is supposed to do now.